### ISSUE TITLE
cftime's calendar type is not passed to the particlefile

### ISSUE THREAD

**Issue by jbusecke** [OceanParcels/parcels#602]
I am using three different xarray datasets to run parcels, all of them use cftime but with different calendars.
I noticed that when saving the parcel results to file, the calendar attribute in the file is `cftime`(see example below), which xarray does not understand upon reading. Is there a way to retain the `calendar` attribute of the original xarray dataset, when using `FieldSet.from_xarray_dataset`?

Example output file:
```
$ ncdump -h eddy_drops_auto_strongest_aviso_seedradius-0.05_0.nc
netcdf eddy_drops_auto_strongest_aviso_seedradius-0.05_0 {
dimensions:
	obs = UNLIMITED ; // (29 currently)
	traj = UNLIMITED ; // (8 currently)
variables:
	int trajectory(traj, obs) ;
		trajectory:long_name = "Unique identifier for each particle" ;
		trajectory:cf_role = "trajectory_id" ;
	double time(traj, obs) ;
		time:_FillValue = NaN ;
		time:long_name = "" ;
		time:standard_name = "time" ;
		time:units = "seconds since 1993-01-01 00:00:00" ;
		time:calendar = "cftime" ;
		time:axis = "T" ;
	float lat(traj, obs) ;
		lat:_FillValue = NaNf ;
		lat:long_name = "" ;
		lat:standard_name = "latitude" ;
		lat:units = "degrees_north" ;
		lat:axis = "Y" ;
	float lon(traj, obs) ;
		lon:_FillValue = NaNf ;
		lon:long_name = "" ;
		lon:standard_name = "longitude" ;
		lon:units = "degrees_east" ;
		lon:axis = "X" ;
	float z(traj, obs) ;
		z:_FillValue = NaNf ;
		z:long_name = "" ;
		z:standard_name = "depth" ;
		z:units = "m" ;
		z:positive = "down" ;

// global attributes:
		:feature_type = "trajectory" ;
		:Conventions = "CF-1.6/CF-1.7" ;
		:ncei_template_version = "NCEI_NetCDF_Trajectory_Template_v2.0" ;
		:parcels_version = "2.0.0b2" ;
		:parcels_mesh = "spherical" ;
}
```

**Comment by erikvansebille** [comment]
The calendar of the ParticleFile object is set at https://github.com/OceanParcels/parcels/blob/master/parcels/particlefile.py#L87

What if you set `ParticleSet.pset.time_origin.calendar = 'standard'`? Does that fix your problem?

**Comment by willirath** [comment]
The non-standard `cftime` date times carry a `.calendar` attribute that has all the info:

```python
import cftime

for cal in [
    'noleap', '365_day', '360_day', 'julian',
    'all_leap', '366_day'
]:
    date = cftime.num2date(
        times=0,
        units="days since 1900-01-01 00:00:00",
        calendar=cal)
    print(f"{cal}: calendar attribute is {date.calendar}")
```

So ideally, `ParticleSet.pset.time_origin.calendar` would be set to `ParticleSet.pset.time_origin.data.calendar` or whatever is necessary to really access the `Datetime` object stored withing `time_origin`.

**Comment by jbusecke** [comment]
That would be ideal @willirath. Then files written by ocean parcels could be 'roundtriped' with xarray (e.g. the time for a loaded file would be exactly the same as the dataset provided to the fieldset).

Looks to me that we could parse the cftime calendar over [here](https://github.com/OceanParcels/parcels/blob/b7e906b626f453ed75d6cb710ba9102a6d0a860e/parcels/tools/converters.py#L25). 
I could see if I can put together a PR for that, or do you think it is more complicated than that?

**Comment by willirath** [comment]
I think one should write a function that wraps all those if clauses leading to self.calendar being set. That would also make it easier to tests this bit. If you could create a PR, I'd leave direct comments there.


Am 7. Juni 2019 17:51:09 schrieb Julius Busecke <notifications@github.com>:

**Comment by jbusecke** [comment]
I took a shot at it in #604. Let me know if you think this is going in the right direction. Happy to modify as needed.

**Comment by erikvansebille** [comment]
This is implemented in v2.0.0
